src/odemis/driver/test/hamamatsurx_test.py

```diff
@@ -0,0 +1,704 @@
+from odemis.driver import hamamatsurx
+from odemis.driver import semcomedi
+import time
+from odemis.driver import andorshrk
+
+import socket
+import numpy
+from matplotlib import pyplot as plt
+import matplotlib.animation as animation
+from odemis.cli.video_displayer import VideoDisplayer
+import threading
+
+import unittest
+from unittest.case import skip
+
+from cam_test_abs import VirtualTestCam, VirtualStaticTestCam, VirtualTestSynchronized  # TODO do we need VirtualStaticTestCam?
+
+logging.getLogger().setLevel(logging.DEBUG)
+
+CLASS = hamamatsurx.StreakCamera
+
+# arguments used for the creation of basic components
+CONFIG_READOUTCAM = {"name": "ReadoutCamera", "role": "readoutcam"}
+CONFIG_STREAKUNIT = {"name": "StreakUnit", "role": "streakunit"}
+CONFIG_DELAYBOX = {"name": "Delaybox", "role": "delaybox"}
+
+children = {"readoutcam": CONFIG_READOUTCAM, "streakunit": CONFIG_STREAKUNIT, "delaybox": CONFIG_DELAYBOX}
+
+KWARGS = dict(name="streak cam", role="ccd", host="DESKTOP-E6H9DJ0", port=1001, children=children)
+
+# test with spectrograph
+CLASS_SHRK = andorshrk.Shamrock
+KWARGS_SHRK_SIM = dict(name="sr193", role="spectrograph", device="fake",
+                       slits={1: "slit-in", 3: "slit-monochromator"},
+                       bands={1: (230e-9, 500e-9), 3: (600e-9, 1253e-9), 5: "pass-through"})
+
+
+class TestHamamatsurxCamWithSpectrograph(unittest.TestCase):
+    """Test the Hamamatsu streak camera class with real streak camera HW and
+     a simulated spectrograph"""
+
+    @classmethod
+    def setUpClass(cls):
+
+        cls.spectrograph = CLASS_SHRK(**KWARGS_SHRK_SIM)
+
+        children = {"readoutcam": CONFIG_READOUTCAM, "streakunit": CONFIG_STREAKUNIT,
+                    "delaybox": CONFIG_DELAYBOX, "spectrograph": cls.spectrograph}
+
+        cls.streakcam = hamamatsurx.StreakCamera("streak cam", "streakcam", host="DESKTOP-E6H9DJ0", port=1001,
+                                                    children=children)
+
+        for child in cls.streakcam.children.value:
+            if child.name == CONFIG_READOUTCAM["name"]:
+                cls.readoutcam = child
+            if child.name == CONFIG_STREAKUNIT["name"]:
+                cls.streakunit = child
+            if child.name == CONFIG_DELAYBOX["name"]:
+                cls.delaybox = child
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.streakcam.terminate()
+
+    def test_magnification(self):
+        """Test the streak lens component and the corresponding magnification is
+        correctly applied to calculate the effective pixel size of the readout camera."""
+
+        # default mag is 1. if not specified
+        wll = self.readoutcam._metadata[model.MD_WL_LIST]
+
+        mag = 0.476
+        md = {model.MD_LENS_MAG: mag}
+        self.readoutcam.updateMetadata(md)
+        self.assertIn(model.MD_LENS_MAG, self.readoutcam._metadata)
+        self.assertEqual(self.readoutcam._metadata[model.MD_LENS_MAG], mag)
```

Comment:
Instead of looking at `._metadata`, use `.getMetadara()`. Admittedly, it's exactly the same result, but that's the official API, so it's better than using some private attribute.